taxi/views.py

```diff
@@ -61,11 +62,22 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def post(self, request, *args, **kwargs):
+        car = self.get_object()
+        action = request.POST.get("action")
+
+        if action == "assign":
+            car.drivers.add(request.user)
+        elif action == "remove":
+            car.drivers.remove(request.user)
+
+        return redirect("taxi:car-detail", pk=car.pk)
```

Comment:
The logic for assigning and removing the current user from the car's drivers list is correctly implemented. However, ensure that the buttons on the car detail page are correctly labeled and functional as per the task requirements.